### PR TITLE
adding upguard_node ansible module

### PR DESCRIPTION
**Pull request by napalm255**
Created base ansible readme framework.
Added 'upguard_node' ansible module.
Added 'play.yml' as a working example.

### CODE REVIEW

File: ansible/upguard_node/library/upguard_node.py [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+            self.module.fail_json(msg='failed querying node group nodes',
+                                  error=str(ex))
+
+        nodes = json.loads(response.content)
+
+        return nodes
+
+    def group_id(self, group_name):
+        """Return group id."""
+        url = '/node_groups/lookup.json?name={}'.format(group_name)
```
Comment (moonpolysoft) [inline ansible/upguard_node/library/upguard_node.py]: we should URL encode `group_name` here.

File: ansible/upguard_node/library/upguard_node.py [hunk 2]
```diff
@@ -0,0 +1,751 @@
...
+        except requests.exceptions.HTTPError as ex:
+            self.module.fail_json(msg='failed adding node to group',
+                                  error=str(ex))
+
+        return added
+
+    @property
+    def node_id(self):
+        """Return node id."""
+        url = '/nodes/lookup.json?name={}'.format(self.arg.name)
```
Comment (moonpolysoft) [inline ansible/upguard_node/library/upguard_node.py]: Same comment as above, we should URL encode `self.arg.name`

File: ansible/upguard_node/library/upguard_node.py [hunk 3]
```diff
@@ -0,0 +1,751 @@
...
+
+        details = json.loads(response.content)
+
+        return details
+
+    @property
+    def exists(self):
+        """Node exists."""
+        node_exists = True
+        url = '/nodes/lookup.json?name={}'.format(self.arg.name)
```
Comment (moonpolysoft) [inline ansible/upguard_node/library/upguard_node.py]: And here.

File: ansible/upguard_node/library/upguard_node.py [hunk 4]
```diff
@@ -0,0 +1,751 @@
...
+
+        return results
+
+    def scan(self):
+        """Scan node."""
+        job = None
+        node_id = self.node_id
+        label = self.arg.scan_label
+        timeout = self.arg.scan_timeout
+        url = '/nodes/{}/start_scan.json?label={}'.format(node_id, label)
```
Comment (moonpolysoft) [inline ansible/upguard_node/library/upguard_node.py]: Same here for label.

**Comment by napalm255** [comment]
It seems the current method does some level of encoding auto-magically... With that said, I think I'll switch up to use the 'params' argument instead, which is probably the better direction anyhow.

>>> a = 'some _string !with $special %chars &to *encode'
>>> 
>>> import requests
>>> r = requests.get('http://www.python.org?name=' + a + '&other=' + a)
>>> r.url
u'https://www.python.org/?name=some%20_string%20!with%20$special%20%25chars%20&to%20*encode&other=some%20_string%20!with%20$special%20%25chars%20&to%20*encode'
>>> r = requests.get('http://www.python.org', params={'name': a, 'other': a})
>>> r.url
u'https://www.python.org/?other=some+_string+%21with+%24special+%25chars+%26to+%2Aencode&name=some+_string+%21with+%24special+%25chars+%26to+%2Aencode'
>>> 
>>> import urllib
>>> urllib.urlencode({'name': a, 'other': a})
'other=some+_string+%21with+%24special+%25chars+%26to+%2Aencode&name=some+_string+%21with+%24special+%25chars+%26to+%2Aencode'
>>>

**Comment by napalm255** [comment]
Changes made and tested.

**Pull request merged**